### PR TITLE
feat: add readOnlyRootFilesystem if possible

### PR DESCRIPTION
**Pull request by farodin91**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
Add containerSecurityContext for containers
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
this change does not work, similar issue: https://github.com/kubernetes-csi/csi-driver-nfs/pull/422

**Comment by farodin91** [comment]
@msau42 updated

**Comment by farodin91** [comment]
@msau42  i re added privileged to the controller

File: charts/latest/csi-driver-smb/values.yaml [hunk 1]
```diff
@@ -138,6 +138,10 @@ windows:
       requests:
         cpu: 10m
         memory: 40Mi
+  containerSecurityContext:
```
Comment (andyzhangx) [inline charts/latest/csi-driver-smb/values.yaml:141]: this PR is different from https://github.com/kubernetes-csi/csi-driver-nfs/pull/422

Comment (farodin91) [reply charts/latest/csi-driver-smb/values.yaml]: removed

**Pull request merged**